cmake/sysbuild/suit.cmake

```diff
@@ -99,6 +99,31 @@ function(suit_generate_dfu_zip)
   )
 endfunction()
 
+# Find a VERSION file path.
+#
+# Usage:
+#   suit_get_version_file(<output_variable>)
+#
+# Parameters:
+#   'output_variable' - variable to store path
+function(suit_get_version_file output_variable)
+  sysbuild_get(CONFIG_DIR IMAGE ${DEFAULT_IMAGE} VAR APPLICATION_CONFIG_DIR CACHE)
+  sysbuild_get(PROJECT_DIR IMAGE ${DEFAULT_IMAGE} VAR APPLICATION_SOURCE_DIR CACHE)
+
+  if(DEFINED VERSION_FILE)
+    set(${output_variable} "${VERSION_FILE}" PARENT_SCOPE)
+    message(STATUS "Using version: ${VERSION_FILE}")
+  elseif((DEFINED CONFIG_DIR) AND (EXISTS ${CONFIG_DIR}/VERSION))
+    set(${output_variable} "${CONFIG_DIR}/VERSION" PARENT_SCOPE)
+    message(STATUS "Using version: ${CONFIG_DIR}/VERSION")
+  elseif((DEFINED PROJECT_DIR) AND (EXISTS ${PROJECT_DIR}/VERSION))
+    set(${output_variable} "${PROJECT_DIR}/VERSION" PARENT_SCOPE)
+    message(STATUS "Using version: ${PROJECT_DIR}/VERSION")
+  else()
+    message(STATUS "Unable to find a VERSION file. Using default values (seq: 1, no semver)")
+  endif()
+endfunction()
```

Comment:
note that this is not compatible with the normal VERSION file, that can only reside in the application source directory, so if someone puts this version file in e.g. the application config dir, it might be used for the suit parts but the version fields will not

Reply:
Discussed offline and decided to remove this extension for now.